sonar-kotlin-plugin/src/test/java/org/sonarsource/kotlin/plugin/KotlinSensorTest.kt

```diff
@@ -208,7 +232,60 @@ internal class KotlinSensorTest : AbstractSensorTest() {
         Assertions.assertThat(analysisErrors).isEmpty()
     }
 
+    private fun failFastSensorWithEnvironmentSetup(failFast: Boolean?): KotlinSensor {
+        mockkStatic("org.sonarsource.kotlin.plugin.KotlinCheckListKt")
+        every { KOTLIN_CHECKS } returns listOf(ExceptionThrowingCheck::class.java)
+
+        context.apply {
+            setSettings(MapSettings().apply {
+                failFast?.let { setProperty("sonar.internal.analysis.failFast", it) }
+            })
+
+            fileSystem().add(createInputFile("file1.kt", "class A { fun f() = TODO() }"))
+        }
+        return sensor(checkFactory("ETRule1"))
+    }
+
+    @Test
+    fun `ensure failFast is working when set`() {
+        val sensor = failFastSensorWithEnvironmentSetup(true)
+
+        assertThrows<IllegalStateException>("Exception in 'KtChecksVisitor' while analyzing 'file1.kt'") {
+            sensor.execute(context)
+        }.apply {
+            assertThat(this).hasCause(TestException("This is a test message"))
+        }
+
+        assertThat(logTester.logs(LoggerLevel.ERROR))
+            .containsExactly("Cannot analyse 'file1.kt' with 'KtChecksVisitor': This is a test message")
+    }
+
+    @Test
+    fun `ensure failFast is not triggering when not set`() {
+        val sensor = failFastSensorWithEnvironmentSetup(false)
+        assertDoesNotThrow { sensor.execute(context) }
+        assertThat(logTester.logs(LoggerLevel.ERROR))
+            .containsExactly("Cannot analyse 'file1.kt' with 'KtChecksVisitor': This is a test message")
+    }
+
+    @Test
+    fun `ensure failFast is not triggered when unset`() {
```

Comment:
Looks like these 2 test names are confused. should these be "not set" -> null and "unset" -> false. Or maybe choose different names to make it less confusing)